core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -434,6 +439,7 @@ public Location add(Operation operation) throws IOException {
                 ensureOpen();
                 Location location = current.add(bytes);
                 assert assertBytesAtLocation(location, bytes);
+                lastWriteLocation = location;
```

Comment:
I wanted to make that lazy too! It felt really nasty to do this. But `Translog.Location` has the size of the last write and I didn't see an easy way to recover that. I could return `Integer.MAX_VALUE` for that size and then it'd always sort after the "real" translog location.